api/src/main/java/org/pmiops/workbench/mailchimp/MailChimpServiceImpl.java

```diff
@@ -58,30 +51,30 @@ public String addUserContactEmail(String contactEmail) {
 
   @Override
   public String getMember(String contactEmail) throws NotFoundException {
-    Map<String, Object> mailchimpResponse;
+    Map<String, Object> mailchimpResponse = new HashMap<String, Object>();
     try {
       mailchimpResponse = getClient().execute(
-          new GetMemberMethod(listId,
-          contactEmail)).mapping;
+          new GetMemberMethod(getListId(),
+              contactEmail)).mapping;
     } catch (MailchimpException e) {
-      if (e.code == 404) {
-        throw new NotFoundException();
-      } else {
-        throw new RuntimeException(e);
-      }
+      throw ExceptionUtils.convertMailchimpError(e);
     } catch (IOException e) {
       throw new RuntimeException(e);
     }
     return mailchimpResponse.get(MailChimpService.MAILCHIMP_KEY_STATUS).toString();
   }
 
+  private String getListId() {
+    if (listId == null) {
+      listId = cloudStorageServiceProvider.get().readMailChimpListId();
```

Comment:
I made a comment to this affect in the original PR - unsychronized concurrent access to this singleton instance variable is setting off alarms for me; though to the best of my Java understanding this shouldn't be a problem - the worst case I think is that you'd read from GCS a couple of extra times (and in an extreme edge case, separate threads might have different values if this changed in GCS).

Flagging this for another look to double check (maybe @danqrodney?) : `this.listId` and `this.apiKey` may be read/written concurrently - shouldn't be a problem here, right?